packages/api/src/controllers/animalBatchController.js

```diff
@@ -149,6 +149,63 @@ const animalBatchController = {
       }
     };
   },
+
+  deleteAnimalBatches() {
+    return async (req, res) => {
```

Comment:
This PR's working great! As a potential improvement, I was thinking that in this particular case the code for deletion for both animals and batches seems to be almost identical, except for the different model they use, so this would be a good case to abstract away into a middleware. I tried it out in a separate branch, `middleware-for-common-code` -- essentially the middleware receives the model and handles the deletion logic. I did have to make the error messages a bit more generic, they could be customized with a parameter too but it didn't seem worth the trouble. Can you take a look and let me know your thoughts?

Reply:
Cool, I love it!!

My only hesitation is with the concept of doing CRUD in middleware itself, because I would have thought that is so completely a controller action. Logically I would lean towards doing the loop that checks on IDs (which is the bulk of the middleware code anyway) in this function (and maybe even extend it to handle the other routes that do the same check on the `req.body` array?) but still let the controller do the actual deletion 🤔 

But if handling the whole CRUD in middleware is a legitimate pattern that I just haven't heard of yet, then I'm fine with fine with giving it a go; it reads very well.

Reply:
Yup that's a really good point, I kinda thought about that too! It is a bit weird to have a middleware without any associated controller method being called after it. I like your idea of abstracting everything except for the actual deletion!